src/state/apis/foxy/foxyBalancesApi.ts

```diff
@@ -0,0 +1,227 @@
+  assets: AssetsById,
+  marketData: MaybeMarketCapData,
+): BN =>
+  Object.values(opportunities).reduce((acc: BigNumber, opportunity: FoxyOpportunity) => {
+    const amount = makeFiatAmount(opportunity, assets, marketData)
+    return acc.plus(bnOrZero(amount))
+  }, bn(0))
+
+const makeMergedOpportunities = (
+  opportunities: Record<string, FoxyOpportunity>,
+  assets: AssetsById,
+  marketData: MaybeMarketCapData,
+) =>
+  Object.values(opportunities).map(opportunity => {
+    const asset = assets[opportunity.tokenAssetId]
+    const fiatAmount = makeFiatAmount(opportunity, assets, marketData)
+    const marketPrice = marketData[opportunity.tokenAssetId]?.price ?? 0
+    const tvl = bnOrZero(opportunity.tvl)
+      .div(`1e+${asset?.precision}`)
+      .times(marketPrice)
+      .toString()
+    const data = {
+      ...opportunity,
+      tvl,
+      cryptoAmount: bnOrZero(opportunity.balance).div(`1e+${asset?.precision}`).toString(),
+      fiatAmount: fiatAmount.toString(),
+    }
+    return data
+  })
+
+async function getFoxyOpportunities(
+  balances: PortfolioBalancesById,
+  api: FoxyApi,
+  userAddress: string,
+  foxyApr: string,
+) {
+  const acc: Record<string, FoxyOpportunity> = {}
+  try {
+    const opportunities = await api.getFoxyOpportunities()
+    for (let index = 0; index < opportunities.length; index++) {
+      // TODO: assetIds in vaults
+      const opportunity = opportunities[index]
+      const withdrawInfo = await api.getWithdrawInfo({
+        contractAddress: opportunity.contractAddress,
+        userAddress,
+      })
+      const rewardTokenAssetId = toAssetId({
+        chainId: opportunity.chain,
+        assetNamespace: 'erc20',
+        assetReference: opportunity.rewardToken,
+      })
+      const contractAssetId = toAssetId({
+        chainId: opportunity.chain,
+        assetNamespace: 'erc20',
+        assetReference: opportunity.contractAddress,
+      })
+      const tokenAssetId = toAssetId({
+        chainId: opportunity.chain,
+        assetNamespace: 'erc20',
+        assetReference: opportunity.stakingToken,
+      })
+      const balance = balances[rewardTokenAssetId]
+
+      const pricePerShare = api.pricePerShare()
+      acc[opportunity.contractAddress] = {
+        ...opportunity,
+        tvl: opportunity.tvl.toString(),
+        apy: foxyApr,
+        chainId: opportunity.chain,
+        balance: bnOrZero(balance).toString(),
+        contractAssetId,
+        tokenAssetId,
+        rewardTokenAssetId,
+        pricePerShare: bnOrZero(pricePerShare).toString(),
+        withdrawInfo,
+      }
+    }
+    return acc
+  } catch (e) {
+    moduleLogger.error(e, { fn: 'getFoxyOpportunities' }, 'Error getting opportunities')
+    return acc
+  }
+}
+
+export const foxyBalancesApi = createApi({
+  reducerPath: 'foxyBalancesApi',
+  // not actually used, only used to satisfy createApi, we use a custom queryFn
+  baseQuery: fetchBaseQuery({ baseUrl: '/' }),
```

Comment:
We can use the built-in helper, `fakeBaseQuery`, here, imported from `'@reduxjs/toolkit/dist/query/react'`:

```suggestion
  baseQuery: fakeBaseQuery(),
```

Reply:
Added in https://github.com/shapeshift/web/commit/886b49b3e6832040d3ce6aa94a9e0ad65d0830d2, very nice find!